turbinia/state_manager.py

```diff
@@ -399,32 +408,42 @@ def write_evidence(self, evidence_dict: dict[str], update=False) -> str:
     Raises:
       TurbiniaException: If the attribute deserialization fails.
     """
+    evidence_key = ''
+    request_key = ''
     try:
       evidence_id = json.loads(evidence_dict['id'])
       evidence_hash = json.loads(evidence_dict.get('hash'))
-      try:
-        # Don't keep the config value since we don't really use it
-        # and it can be quite verbose if a complex recipe is used.
-        evidence_dict.pop('config')
-      except KeyError:
-        # Nothing to do if the key doesn't exist.
-        pass
-      request_key = ':'.join(
-          ('TurbiniaRequest', json.loads(evidence_dict['request_id'])))
+      request_id = json.loads(evidence_dict['request_id'])
+      request_key = self.redis_client.build_key_name('request', request_id)
+      evidence_key = self.redis_client.build_key_name('evidence', evidence_id)
     except (TypeError, ValueError) as exception:
       error_message = 'Error deserializing evidence attribute.'
       log.error(f'{error_message}: {exception}')
       raise TurbiniaException(error_message) from exception
-    evidence_key = ':'.join(('TurbiniaEvidence', evidence_id))
-    # Either updates or writes new key
-    if update == self.redis_client.key_exists(evidence_key):
-      self.redis_client.write_hash_object(evidence_key, evidence_dict)
-      if evidence_hash:
-        self.redis_client.set_attribute(
-            'TurbiniaEvidenceHashes', evidence_hash, evidence_key)
-      if not update:
-        self.redis_client.add_to_list(request_key, 'evidence_ids', evidence_id)
-      return evidence_key
+
+    # Don't keep the config value since we don't really use it
+    # and it can be quite verbose if a complex recipe is used.
+    try:
+      evidence_dict.pop('config')
+    except KeyError:
+      # Nothing to do if the key doesn't exist.
+      pass
+
+    try:
+      # Either updates or writes new key
+      if update == self.redis_client.key_exists(evidence_key):
```

Comment:
I'm still not sure I get this logic.  If update==True and the key does not exist, then we still won't write the key.   Maybe I'm missing something?